colin-api/src/colin_api/resources/business.py

```diff
@@ -170,3 +170,31 @@ def get(info_type, legal_type=None, identifier=None):  # pylint: disable = too-m
         except Exception as err:  # pylint: disable=broad-except; want to catch all errors
             current_app.logger.error(err.with_traceback(None))
             return jsonify({'message': 'Something went wrong.'}), HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+@cors_preflight('GET')
+@API.route('<string:corp_num>', methods=['GET'])
+class BusinessAllInfo(Resource):
+    """Meta information about the overall service."""
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @conditional_auth(jwt.requires_roles, [COLIN_SVC_ROLE])
+    def get(corp_num: str):
+        """Return the complete business info."""
+        try:
+            # get business
+            business = Business.find_by_corp_num(corp_num)
+            if not business:
+                return jsonify({'message': f'{corp_num} not found'}), HTTPStatus.NOT_FOUND
+            return jsonify(business.as_dict(slim=True)), HTTPStatus.OK
+
+        except GenericException as err:  # pylint: disable=duplicate-code
+            return jsonify({'message': err.error}), err.status_code
```

Comment:
any `BusinessNotFoundException` thrown by `find_by_corp_num` should land in here resulting in a 404 being returned to api consumer.

Reply:
I updated